File: src/models/Ride.ts
import { Location } from './Driver';

/**
 * Ride model representing a carrera
 */
export interface Ride {
  id: string;
  clientId: string;
  originAddress: string;
  destinationAddress: string;
  originLocation: Location;
  destinationLocation: Location;
  estimatedAmount: number;
  distanceFromDriver: number; // in kilometers
  tripDistance: number; // in kilometers
  status: RideStatus;
  createdAt: Date;
  otp?: string;
}

export enum RideStatus {
  AVAILABLE = 'available',
  ACCEPTED = 'accepted',
  IN_PROGRESS = 'in_progress',
  COMPLETED = 'completed',
  CANCELLED = 'cancelled',
}

File: app/src/main/java/com/motodriver/app/data/model/Ride.kt
package com.motodriver.app.data.model

import java.util.Date

/**
 * Ride status enum representing the current state of a carrera
 */
enum class RideStatus {
    AVAILABLE,
    ACCEPTED,
    IN_PROGRESS,
    COMPLETED,
    CANCELLED
}

/**
 * Ride model representing a carrera
 */
data class Ride(
    val id: String,
    val clientId: String,
    val originAddress: String,
    val destinationAddress: String,
    val originLocation: Location,
    val destinationLocation: Location,
    val estimatedAmount: Int,
    val distanceFromDriver: Double, // in kilometers
    val tripDistance: Double, // in kilometers
    val status: RideStatus,
    val createdAt: Date,
    val otp: String? = null
)
